### PR TITLE
fix: Detect insulators and simulators in mythic found feature [skip ci]

### PR DESCRIPTION
**Pull request by ShadowCat117**
They are technically mythics so should be counted

### CODE REVIEW

**Comment by ShadowCat117** [comment]
Also I should note, I think unid shiny items are parsed as `UnknownGearItem` so the feature probably won't detect those either. I did try fixing it a while ago but couldn't get a good regex pattern for it

**Comment by kristofbolyai** [comment]
> Also I should note, I think unid shiny items are parsed as `UnknownGearItem` so the feature probably won't detect those either. I did try fixing it a while ago but couldn't get a good regex pattern for it

Perhaps make an issue?

File: common/src/main/java/com/wynntils/features/MythicFoundFeature.java [hunk 1]
```diff
@@ -52,11 +54,25 @@ public void onMythicFound(MythicFoundEvent event) {
...
+            sendLootrunDryStreakMessage(
+                    StyledText.fromComponent(event.getMythicBoxItem().getHoverName()));
+            return;
+        }
+
+        Optional<SimulatorItem> simulatorItem = Models.Item.asWynnItem(itemStack, SimulatorItem.class);
+        if (simulatorItem.isPresent()) {
+            sendLootrunDryStreakMessage(
+                    StyledText.fromComponent(event.getMythicBoxItem().getHoverName()));
         }
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/MythicFoundFeature.java]: Is this the only place this check is used? Does the mythic found event fire for these items?

Comment (ShadowCat117) [reply common/src/main/java/com/wynntils/features/MythicFoundFeature.java]: Oops yeah that does need it too

File: common/src/main/java/com/wynntils/models/lootrun/LootrunModel.java [hunk 2]
```diff
@@ -307,11 +309,26 @@ public void onEntitySpawn(SetEntityDataEvent event) {
...
+                    dryPulls.store(0);
+                    return;
+                }
 
-                if (gearItem.getGearTier() == GearTier.MYTHIC) {
+                Optional<SimulatorItem> simulatorItemOpt = Models.Item.asWynnItem(itemStack, SimulatorItem.class);
+                if (simulatorItemOpt.isPresent()) {
                     WynntilsMod.postEvent(new MythicFoundEvent(itemStack, true));
                     dryPulls.store(0);
                 }
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/models/lootrun/LootrunModel.java]: Repeat yourself twice, but the third time :). Can you add a "mythicFound" boolean and do the remaining work there? Maybe do this in the feature too.

**Review by kristofbolyai — APPROVED** [review]

**Pull request merged**